app/models/mapping.rb

```diff
@@ -149,63 +158,44 @@ def build_facl_entry_for_user(user, domain)
     )
   end
 
-  # Determine if FACLs can be set by the admin user
-  #
-  # Data sets may be stored in non-standard locations, and oqwned by users
-  # other than the admin user. In the case of a non-owned dataset a mapping
-  # may be valid, but the admin user will not be able to set FACLS, and
-  # should not try.
-  def can_change_facls?(path)
-    owns = false
+  # Check if a RX ACL entry exists on pathname
+  # @return [Boolean]
+  def rx_facl_exists?(pathname)
     begin
-      owns = File.stat(path).owned?
+      acl = OodSupport::ACLs::Nfs4ACL.get_facl(path: pathname)
+      expected = build_facl_entry_for_user(user, FACL_USER_DOMAIN)
+
+      return acl.entries.include?(expected)
     rescue
+      return false
     end
-
-    owns
   end
 
-  # Add user FACLs to app and dataset
-  # @return errors [Exception]
-  def add_user_facls(mapping)
-    absolute_app_path = File.join(
-      File.expand_path(ENV['APP_PROJECT_SPACE']),
-      mapping.app
-    )
+  # Check whether a user has read/execute permissions on the app and dataset directories
+  # @return [Boolean] does user have correct permissions?
+  def user_has_permissions_on_both
```

Comment:
Is this still a necessary method now that we have `rx_facl_exists?`? Seems like the body of this method duplicates the purpose of `rx_facl_exists?`.

Reply:
I'd like to leave it in as a shortcut, but calling `Mapping.rx_facl_exists?` instead of duplicating the work. Also since it's a Boolean method I'll add a question mark.

Reply:
Actually I spoke to soon. They do different things. `user_has_permissions_on_both` ensures that a user is `allow?'d` to do something, whereas `rx_facl_exists?` checks for the entry. A user can have permission to do something without a FACL entry existing.

Reply:
Oh also when referring to instance methods of a class, use `Mapping#rx_facl_exists?` instead of `Mapping.rx_facl_exists?` as the `Mapping.rx_facl_exists?` form by convention refers to the "class method" instead of the "instance method".

Reply:
Noted.

Reply:
And on third thought I am going to re-write `user_has_permissions_on_both?` to use the Pathname tests as they do not throw.

Reply:
...except that the Pathname methods only test for properties when acting as `$USER`, so we do need the flexibility of the FACL testing.